### PR TITLE
tests: add more cases for oopif smoke test

### PR DESCRIPTION
**Pull request by connorjclark**
I didn't understand under what circumstances we determine a script (via `ScriptElement`) to be from an OOPIF (when the network record has a sessionId thus is not from the main target).

Learned some things:

- iframes from the same domain (even different ports) will be in the same target, thus don't get filtered
- but workers are a different target (with a sessionId) so it's considered an "OOPIF" and all its scripts are blocked
- ditto for service workers (but I didn't bother adding a test case for that)

Adding more test cases to capture how things change if we decide to stop filtering "OOPIF" scripts from our audits.

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            _includes: [
+              {url: 'http://localhost:10200/oopif-scripts.html'},
+              {url: 'http://localhost:10200/oopif-simple-page.html'},
+              {url: 'http://localhost:10503/oopif-simple-page.html'},
+              {url: 'http://localhost:10200/simple-script.js'},
+              {url: 'http://localhost:10503/simple-script.js'},
+              {url: 'http://localhost:10200/simple-script.js'},
+              {url: 'http://localhost:10503/simple-script.js'},
+              {url: 'http://localhost:10200/simple-worker.js'},
+              {url: 'http://localhost:10503/simple-worker.js'},
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js:58]: for some reason simple-worker.mjs never showed up....

Comment (connorjclark) [reply lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js:61]: that's fine! I just want to document things as they are now.

I added a new `_excludes` assertion so we will know when this stops happening with future refactors

File: lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  artifacts: {
+    IFrameElements: [
+      {
+        id: 'iframe-1',
+        src: 'http://localhost:10200/oopif-simple-page.html',
+        clientRect: {
+          width: '>0',
+          height: '>0',
+        },
+        isPositionFixed: true,
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js:82]: current test failure: this should be true but smokes is seeing false....

File: lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      'network-requests': {
+        details: {
+          items: {
+            _includes: [
+              {url: 'http://localhost:10200/oopif-scripts.html'},
+              {url: 'http://localhost:10200/oopif-simple-page.html'},
+              {url: 'http://localhost:10503/oopif-simple-page.html'},
+              // simple-script.js is included many times
+              // 2 * (1 from <script>, 1 from fetch) = 4
+              // Note, the network records from the workers are _not_ captured!
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js:55]: bug alert! ⚠️

File: lighthouse-cli/test/smokehouse/report-assert.js [hunk 4]
```diff
@@ -140,6 +144,34 @@ function findDifference(path, actual, expected) {
...
 
+    if (key === '_excludes') {
+      // Re-use state from `_includes` check, if there was one.
+      /** @type {any[]} */
+      const arrToCheckAgainst = inclExclCopy || actual;
+
+      if (!Array.isArray(expectedValue)) throw new Error('Array subset must be array');
+      if (!Array.isArray(actual)) continue;
+
+      const unexpectedValues = expectedValue;
```
Comment (adamraine) [inline lighthouse-cli/test/smokehouse/report-assert.js:155]: Lol @ expected -> unexpected. Could we make this `expectedExclusions` or something?

File: lighthouse-cli/test/smokehouse/report-assert.js [hunk 5]
```diff
@@ -140,6 +144,34 @@ function findDifference(path, actual, expected) {
...
+      console.log({arrToCheckAgainst});
+      for (const unexpectedEntry of unexpectedValues) {
+        const matchingIndex = arrToCheckAgainst.findIndex(actualEntry =>
```
Comment (adamraine) [inline lighthouse-cli/test/smokehouse/report-assert.js:158]: We don't do any splicing, can probably just use `arrToCheckAgainst.find`

File: lighthouse-cli/test/smokehouse/readme.md [hunk 6]
```diff
@@ -66,6 +66,8 @@ However, if an array literal is used as the expectation, an extra condition is e
 
 Arrays can be checked against a subset of elements using the special `_includes` property. The value of `_includes` _must_ be an array. Each assertion in `_includes` will remove the matching item from consideration for the rest.
 
+Arrays can be asserted to not match any elements using the special `_excludes` property. The value of `_excludes` _must_ be an array. If an `_includes` check is defined before an `_excludes` check, only the element not matched under the previous will be considered.
```
Comment (adamraine) [inline lighthouse-cli/test/smokehouse/readme.md:69]: Can you add an example showing the exhaustive includes case?

File: lighthouse-cli/test/smokehouse/report-assert.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-cli/test/smokehouse/report-assert.js:158]: hm... but then there's the case of `undefined` being a valid value in these arrays :P

Comment (adamraine) [reply lighthouse-cli/test/smokehouse/report-assert.js:157]: Ah ya, index works fine then

**Review by adamraine — APPROVED** [review]

File: lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-cli/test/smokehouse/test-definitions/oopif-scripts.js:61]: :pikaface:

I just tried with ToT chrome and .mjs showed up...

**Pull request merged**